src/netaddress.cpp

```diff
@@ -1068,15 +1101,23 @@ CSubNet::CSubNet(const CNetAddr& addr, const CNetAddr& mask) : CSubNet()
 
 CSubNet::CSubNet(const CNetAddr& addr) : CSubNet()
 {
-    valid = addr.IsIPv4() || addr.IsIPv6();
-    if (!valid) {
+    switch (addr.m_net) {
+    case NET_IPV4:
+    case NET_IPV6:
+        valid = true;
+        memset(netmask, 0xFF, sizeof(netmask));
+        break;
+    case NET_ONION:
+    case NET_I2P:
+    case NET_CJDNS:
+        valid = true;
+        break;
+    case NET_INTERNAL:
+    case NET_UNROUTABLE:
+    case NET_MAX:
         return;
     }
 
-    assert(addr.m_addr.size() <= sizeof(netmask));
```

Comment:
Should we keep this `assert`?
```diff
     case NET_IPV4:
     case NET_IPV6:
         valid = true;
+        assert(addr.m_addr.size() <= sizeof(netmask));
         memset(netmask, 0xFF, sizeof(netmask));
         break;
     case NET_ONION:
```

Reply:
The assert was there because in `master` we wrote `addr.m_addr.size()` bytes and here it is changed to write `sizeof(netmask)` bytes so it can't overflow and thus I removed the assert.

However, I changed/restored it to write `addr.m_addr.size()` bytes and also restored the assert, [explanation](https://github.com/bitcoin/bitcoin/pull/20852#issuecomment-755393067).